src/main/java/jpa/dao/CompanyDao.java

```diff
@@ -0,0 +1,46 @@
+package jpa.dao;
+
+import jpa.entity.Company;
+import lombok.extern.log4j.Log4j;
+
+import javax.persistence.EntityManager;
+
+@Log4j
+public class CompanyDao implements Contract<Company> {
```

Comment:
Create for each entity its own interface that extended from Contract (please change this name).
And these interfaces should be implemented by your classes
CompanyDao ->CompanyDaoImpl
```
public interface CompanyDao extends Contract<Company>{
// if you need, add specific methods
} 
```